src/lib/spinel/spinel.h

```diff
@@ -343,7 +343,7 @@
 // ----------------------------------------------------------------------------
 
 #define SPINEL_PROTOCOL_VERSION_THREAD_MAJOR 4
-#define SPINEL_PROTOCOL_VERSION_THREAD_MINOR 3
+#define SPINEL_PROTOCOL_VERSION_THREAD_MINOR 4
```

Comment:
I think we should not change the `SPINEL_PROTOCOL_VERSION_THREAD_MINOR`.

- The policy for spinel is to ensure backward compatibility when adding new fields and properties (i.e. newer NCP image should stay compatible with older host side drive).
https://github.com/openthread/openthread/blob/f8f75d6d05064fd63e81a3ad4518e5b3728303f8/src/lib/spinel/spinel.h#L295-L306

- We don't have a explicit policy for updating the spinel version number, but the idea for it is that it would change when there is bigger change that does not gurantee backward comatibility.
- **Changing the version can add new restrictions in product integration and firmware update (between host firmware and NCP/RCP firmware ).**


Here is the challenge with Posix/RCP model if we change the `VERSION_THREAD_MINOR`.
- The host code (radio spinel) does check the version of RCP to match its own version and otherwise it would exit with an error.
https://github.com/openthread/openthread/blob/f8f75d6d05064fd63e81a3ad4518e5b3728303f8/src/lib/spinel/radio_spinel_impl.hpp#L232-L252
- So with this existing code, changing the VERSION_MINOR would mean, whenever POSIX code is updated on host side, we need to also update the RCP firmware along with it.

We have similar check in `wpantund` which checks the version (and also in other internal drivers we use on products (RTOS driver)) which mean we would need to update `wpantund` on host with any change on NCP firmware (which is not desirable).

Reply:
I see. I will add the new fields at the end of the spinel to maintain backward compatibility. However, I don't quite like current implementation of TxDone metadata. It is appended at the end of `SPINEL_PROP_LAST_STATUS`. It is not well organized and not easy to add new info. Maybe we can consider changing it later when the version of spinel protocol is updated someday.

Reply:
Me too. I very much dislike the use of `LAST_STATUS` for tx done as well. Mentioned the same here https://github.com/openthread/openthread/pull/4976#discussion_r433343963. (I would be very happy if/when we change/re-design this in future... :)...).